contracts/src/v0.8/dev/KeeperRegistryDev.sol

```diff
@@ -134,14 +134,42 @@ contract KeeperRegistryDev is
 
     uint256 height = block.number;
     if (!isOwner) {
-      height = height + CANCELATION_DELAY;
+      height = height + CANCELLATION_DELAY;
     }
     s_upkeep[id].maxValidBlocknumber = uint64(height);
     s_upkeepIDs.remove(id);
 
     emit UpkeepCanceled(id, uint64(height));
   }
 
+  /**
+   * @notice pause an upkeep
+   * @param id upkeep to be paused
+   */
+  function pauseUpkeep(uint256 id) external override onlyActiveUpkeep(id) onlyOwnerOrRegistrar {
```

Comment:
I believe the `onlyActiveUpkeep` refers here to "non-cancelled" upkeep.
Can we make the distinction between paused and cancelled clearer to prevent confusion

Reply:
I decided to add 2 more errors:
OnlyPausedUpkeep and OnlyUnpausedUpkeep
This will keep the original semantics of "active" the same, which means not cancelled. In the meantime, introduce a new semantics of paused/nonpaused upkeeps.

technically, now "getActiveUpkeepIDs" should be "getActiveNonPausedUpkeepIDs" but i think it's not worth changing the interface for now.

Reply:
Can we maybe change `onlyActiveUpkeep` to `onlyNonCancelledUpkeep` or something similar? 
Basically have distinct semantics of paused and cancelled, and consider active to be both non paused and non cancelled

Reply:
Done